src/add.rs

```diff
@@ -0,0 +1,114 @@
+use std::path::Path;
+
+use std::fs;
+use toml_edit::{Array, DocumentMut, Formatted, Value};
+
+use crate::{config::ConfigLoadError, Config};
+
+pub fn read_and_verify_config(config_file: impl AsRef<Path>) -> Result<DocumentMut, AddError> {
+    let config_file = config_file.as_ref();
+    let _ = Config::from_file(config_file).map_err(|e| AddError{
+        path: config_file.into(),
+        source: AddErrorKind::ConfigLoad(e)
+    })?;
+    let config_content = fs::read_to_string(&config_file).map_err(|e| AddError {
```

Comment:
you can unwrap here and L21 since if you are able to load the config that means that the config file exists and is valid TOML